### PR TITLE
Add 'date_part' function to PostgresModule

### LINKED ISSUE #206: Add 'date_part' function to PostgresModule
Add support for function `date_part` described in the PostgreSQL [docs](https://www.postgresql.org/docs/current/functions-datetime.html).
1. Add function to `PostgresFunctionDef` in `PostgresModule`
2. Add test in `FunctionDefSpec`

### PR DESCRIPTION
**Pull request by riccardocorbella**
Fixes #206

### CODE REVIEW

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 1]
```diff
@@ -108,7 +108,10 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
-    implicit def literal[A: TypeTag](a: A): Expr[Features.Literal, Any, A] = Expr.Literal(a)
+    implicit def literal[A](a: A)(implicit typeTag: TypeTag[A]): Expr[Features.Literal, Any, A] = typeTag match {
+      case TypeTag.TInstant => Expr.literal(s"TIMESTAMP '$a'").asInstanceOf[Expr[Features.Literal, Any, A]]
```
Comment (robmwalsh) [inline core/jvm/src/main/scala/zio/sql/expr.scala:112]: Can you revert this change please? see comment at https://github.com/zio/zio-sql/pull/307#discussion_r528286267

**Review by robmwalsh — CHANGES_REQUESTED** [review]
Thanks for your work on this! I think this should work with proper rendering of literals but unfortunately this will need to wait for issue #311 to be resolved (which will break your literal trick, as this will be escaped to an SQL string)

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 2]
```diff
@@ -108,7 +108,9 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
-    implicit def literal[A: TypeTag](a: A): Expr[Features.Literal, Any, A] = Expr.Literal(a)
+    implicit def literal[A](a: A)(implicit typeTag: TypeTag[A]): Expr[Features.Literal, Any, A] = typeTag match {
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/expr.scala:111]: Let's go back to the initial shorter form ;)

**Comment by jczuchnowski** [comment]
@riccardocorbella just one small fix and we're good to go!

**Comment by riccardocorbella** [comment]
@jczuchnowski #311 is still open, I think that I'm blocked unless I use the version of literal that @robmwalsh asked me to revert above...

**Comment by robmwalsh** [comment]
@riccardocorbella I think that's ready to go now, just need @jczuchnowski to confirm he's happy with it to merge, then we should be able to get this one in

**Comment by jczuchnowski** [comment]
@riccardocorbella https://github.com/zio/zio-sql/pull/317 has been merged

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**